prov/rxm/src/rxm_ep.c

```diff
@@ -154,11 +154,14 @@ static int rxm_buf_reg(void *pool_ctx, void *addr, size_t len, void **context)
 
 			switch (pool->type) {
 			case RXM_BUF_POOL_TX_MSG:
+				tx_buf->pkt.hdr.flags = FI_MSG;
+				/* fall through */
 			case RXM_BUF_POOL_RMA:
 				tx_buf->pkt.ctrl_hdr.type = ofi_ctrl_data;
 				tx_buf->pkt.hdr.op = ofi_op_msg;
 				break;
 			case RXM_BUF_POOL_TX_TAGGED:
+				tx_buf->pkt.hdr.flags = FI_TAGGED;
```

Comment:
We would need to update the protocol version for this change.

Reply:
I think the protocol has already been updated for the next release.

Reply:
Yep, right. The `RXM_OP_VERSION` value has been updated, when I've added the protocol changes in #4109.